serializable.ts

```diff
@@ -40,6 +40,11 @@ export declare interface Serialize {
   tsSerialize(): JSONObject;
 }
 
+/** deep copy `this` */
+export declare interface Clone {
+  clone(jsonObject: Partial<this>): this;
+}
+
```

Comment:
Is this interface being used anywhere? Is this different from the method on `Serializable`?

Reply:
I made these interfaces for implementing if authors wanted to, thinking about it now though I am unsure if we need clone...

Reply:
keeping for now, we can chat about it before v2 release